site-src/concepts/security-considerations.md

```diff
@@ -0,0 +1,174 @@
+# Gateway API security considerations
+
+Gateway controllers can be deployed in a multi-tenant environment, where different
+namespaces are used by different users and customers.
+
+Some caution should be taken by the cluster administrators and Gateway owners to
+provide a safer environment.
+
+## Avoiding hostname/domain hijacking
+
+Gateway controllers work to disambiguate and detect conflicts caused by sharing
+different ports, protocols, etc. between various listeners. (?)
+
+Generally this conflict detection works on a first-come, first-served basis, where
+the first created resource wins in the  conflict management.
+
+The [hostname definition](../reference/spec.md#httproutespec) is a list, so given the
+following scenario:
+
+* `Gateway` accepts routes from a set of namespaces
+* `HTTPRoute` with name `route1` is created on namespace `ns1`, with `creationTimestamp: 00:00:01`
+and defines hostname `something.tld`.
+* `HTTPRoute` with name `route2` is created on namespace `ns2`, with `creationTimestamp: 00:00:30`
+and defines hostname `otherthing.tld`.
+
+The owner of `route1` can hijack the domain `otherthing.tld` from `route2` because
+`route1` is an older resource.
+
+To avoid this situation, the following actions should be taken:
+
+* On shared gateways, admin SHOULD specify on the listener definitions different
+domains, and specific namespaces allowed to use each domain, as the example below:
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: gateway1
+spec:
+  listeners:
+  - hostname: "something.tld"
+    port: 80
+    protocol: HTTP
+    allowedRoutes:
+      namespaces:
+        from: Selector
+        selector:
+            kubernetes.io/metadata.name: ns1
+```
+
+* Because the number of listeners is limited on a Gateway, the administrator should
+instead rely on some validation (like ValidationAdmissionPolicy or some other mechanism)
+that limits what hostnames can be used on routes of each namespaces.
```

Comment:
I don't think this should be a default recommendation. Maybe "If you need more listeners than a single Gateway can support, you should consider...".